trade-finance-manager-ui/server/controllers/case/cancellation/bank-request-date.controller.ts

```diff
@@ -17,19 +20,43 @@ export type PostBankRequestDateRequest = CustomExpressRequest<{
  * @param req - The express request
  * @param res - The express response
  */
-export const getBankRequestDate = (req: GetBankRequestDateRequest, res: Response) => {
+export const getBankRequestDate = async (req: GetBankRequestDateRequest, res: Response) => {
   const { _id } = req.params;
-  const { user } = asUserSession(req.session);
+  const { user, userToken } = asUserSession(req.session);
 
-  // TODO DTFS2-7409: Check deal type allows for cancellation and deal hasn't already been cancelled
+  try {
+    const deal = await api.getDeal(_id, userToken);
 
-  const bankRequestDateViewModel: BankRequestDateViewModel = {
-    activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.ALL_DEALS,
-    user,
-    ukefDealId: '0040613574', // TODO DTFS2-7409: get values from database
-    dealId: _id,
-  };
-  return res.render('case/cancellation/bank-request-date.njk', bankRequestDateViewModel);
+    if (!deal || 'status' in deal) {
+      return res.redirect('/not-found');
+    }
+
+    if (!canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType)) {
+      return res.redirect(`/case/${_id}/deal`);
+    }
+
+    const cancellation = await api.getDealCancellation(_id, userToken);
+
+    const previouslyEnteredBankRequestDate = cancellation?.bankRequestDate ? new Date(cancellation.bankRequestDate) : undefined;
+
+    const day = previouslyEnteredBankRequestDate ? format(previouslyEnteredBankRequestDate, 'd') : '';
+    const month = previouslyEnteredBankRequestDate ? format(previouslyEnteredBankRequestDate, 'M') : '';
+    const year = previouslyEnteredBankRequestDate ? format(previouslyEnteredBankRequestDate, 'yyyy') : '';
```

Comment:
Same comment as above

Reply:
Have implemented above, but wouldn't work here - we explicitly want day/month/year to be an empty string to be passed in the nunjucks template in the case that there isn't a previously entered bank request date

Reply:
Makes sense, thanks 👍